pkg/server/middleware/middleware.go

```diff
@@ -74,14 +74,17 @@ type contextVals struct {
 
 type key int
 
-const tokenKey key = iota
-const GRPCAuthMetadataKey = "grpc-auth"
+const (
+	tokenKey               key = iota
+	GRPCAuthMetadataKey        = "grpc-auth"
+	GitProviderTokenHeader     = "X-Git-Provider-Token"
```

Comment:
https://datatracker.ietf.org/doc/html/rfc6648

Reply:
sounds good, I'll remove the `X-` prefix